examples/lib/src/main/java/momento/client/example/util/AuthUtil.java

```diff
@@ -1,21 +1,59 @@
 package momento.client.example.util;
 
 import momento.sdk.auth.CredentialProvider;
-import momento.sdk.auth.EnvVarCredentialProvider;
 import momento.sdk.exceptions.SdkException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.secretsmanager.SecretsManagerClient;
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueRequest;
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueResponse;
 
 /** Small utility class to vend credentials */
 public class AuthUtil {
-  private static final String AUTH_TOKEN_ENV_VAR = "MOMENTO_AUTH_TOKEN";
+  private static final String AUTH_TOKEN_VAR = "MOMENTO_AUTH_TOKEN";
   private static final Logger logger = LoggerFactory.getLogger(AuthUtil.class);
 
-  public static CredentialProvider getCredentials() {
+  /** Helper method to vend Momento credentials through auth token stored in environment variable */
+  public static CredentialProvider getCredentialsFromEnvironmentVariable() {
     try {
-      return new EnvVarCredentialProvider(AUTH_TOKEN_ENV_VAR);
+      return CredentialProvider.fromEnvVar(AUTH_TOKEN_VAR);
     } catch (SdkException e) {
-      logger.error("Unable to load credential from environment variable " + AUTH_TOKEN_ENV_VAR, e);
+      logger.error("Unable to load credential from environment variable " + AUTH_TOKEN_VAR, e);
+      throw e;
+    }
+  }
+
+  /** Helper method to vend Momento credentials through auth token stored in secrets manager */
+  public static CredentialProvider getCredentialsFromSecretsManagerAuthToken() {
```

Comment:
thanks! I'm super excited to have a secrets manager example! let's talk tomorrow about how we should structure the examples to fit this in though.

Reply:
Yep I looked at your comment on the other PR after I posted this haha. Have replied their but we can sync tomorrow.